pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/DeletedObjectDao.java

```diff
@@ -0,0 +1,201 @@
+package org.broadinstitute.dsm.db.dao;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.structure.DBElement;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.model.patch.DeleteType;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+public class DeletedObjectDao {
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_TISSUE = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)), ') from ddp_tissue where tissue_id = ? ;') into @sql "
+            + " from information_schema.columns where table_name = 'ddp_tissue'";
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_SM_ID = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)), ') from sm_id where sm_id_pk = ? ;') into @sql "
+            + " from information_schema.columns where table_name = 'sm_id'";
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_ONC_HISTORY = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)),') from ddp_onc_history_detail where onc_history_detail_id = ? ;')"
+            + " into @sql from information_schema.columns where table_name = 'ddp_onc_history_detail'";
+    
+    private static final String TABLE_PLACE_HOLDER = "$table";
+    private static final String PK_PLACE_HOLDER = "$pk";
+
+    private static final String SQL_DELETE_VALUES = "DELETE FROM $table WHERE $pk = ?";
+    private static final String SQL_INSERT_DELETED_VALUES = "INSERT INTO deleted_object SET original_table = ?, original_primary_key = ? ,"
+            + " data =?, deleted_by = ?,  deleted_date = ? ";
+
+    private static final String SQL_GET_DELETED_OBJECT_BY_ORIGINAL_ID_TABLE = "SELECT data FROM deleted_object WHERE original_table = ? "
+            + " AND original_primary_key = ? ";
+
+    public static boolean deletePatch(@NonNull String id, @NonNull String user,
+                                      @NonNull DBElement dbElement, DeleteType deleteType) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            //Select everything in the db as the current state of the object into a json string , this is better than creating a Dto because
+            // this can be done independent of class
+            String currentData = getJsonRepOfDeletedObject(conn, id, deleteType);
```

Comment:
I think the transaction management here looks right--since we are using sql vars in this process, it's critical that we're not opening up separate transactions and setting `@sql`, since doing so could wipe out (or mis-set) `@sql` values across web requests.  I _think_ we're safe, but I'm flagging this as a concern if future editors split transactions in the wrong place.  Might be worth a comment or two and some warnings in the queries that use `@sql`.

Reply:
Yes the inserting the json of the we want to delete and deleting it from its original table should all be done in one transaction, I will leave a comment explaining it